termuic/termuiConsole.go

```diff
@@ -0,0 +1,276 @@
+package termuic
+
+import (
+	"os"
+	"os/signal"
+	"strings"
+	"sync"
+	"syscall"
+	"time"
+
+	"github.com/ElrondNetwork/elrond-go/core"
+	ui "github.com/gizak/termui/v3"
+)
+
+// TermuiConsole data where is store data from handler
+type TermuiConsole struct {
+	TermuiConsoleMetrics sync.Map
+	logLines             []string
+	mutLogLineWrite      sync.RWMutex
+	viewBigLog           bool
+	numLogLines          int
+	termuiConsoleWidgets *termuiConsoleGrid
+}
+
+//NewTermuiConsole method is used to return a new TermuiConsole structure
+func NewTermuiConsole() *TermuiConsole {
+
+	tc := TermuiConsole{}
+
+	tc.initMetricsMap()
+
+	tc.viewBigLog = false
+	tc.numLogLines = 10
+
+	return &tc
+}
+
+func (tc *TermuiConsole) Write(p []byte) (n int, err error) {
+
+	go func() {
+		logLine := string(p)
+
+		tc.mutLogLineWrite.Lock()
```

Comment:
Avoid doing 2 atomic operations in 2 different critical sections. You should have prepared the data (logLine) and then do the slice trim and append in one critical section protected by mutLogLineWrite

Reply:
done